sklearn/pipeline.py

```diff
@@ -305,6 +305,13 @@ def predict(self, X):
             Data to predict on. Must fulfill input requirements of first step
             of the pipeline.
 
+        **predict_params : dict of string -> object
+            Parameters passed to the final ``predict`` in the pipeline. Note
```

Comment:
I find the explanation a bit cryptic. It says it's only passed to the last step, right? Incidentally: why do we have that and not ``transform_params``? And if we had ``transform_params`` would they be passed to all steps or only the last?

Separately, I might have preferred using a different method instead of having ``return_...`` in ``predict``, but I guess that ship is sailed. I don't remember the argument for this.

Apart from the somewhat cryptic docstring lgtm, as it is a logical consequence of our current GP API.

Reply:
I believe that if we had `transform_params` they would need to have be passed to all intermediate steps, yes. In this case, `predict_params` is passed for the only call of predict in the pipeline, which happens to be the last one. I tried making the docstring a little bit more clear. Thanks for the suggestion!